src/pykeen/nn/modules.py

```diff
@@ -0,0 +1,1092 @@
+# -*- coding: utf-8 -*-
+
+"""Stateful interaction functions."""
+
+from __future__ import annotations
+
+import logging
+import math
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Generic, Mapping, MutableMapping, Optional, Sequence, TYPE_CHECKING, Tuple, Union
+
+import torch
+from torch import FloatTensor, nn
+
+from . import functional as pkf
+from ..typing import HeadRepresentation, RelationRepresentation, TailRepresentation
+from ..utils import CANONICAL_DIMENSIONS, convert_to_canonical_shape, ensure_tuple, upgrade_to_sequence
+
+if TYPE_CHECKING:
+    from ..typing import Representation  # noqa
+
+__all__ = [
+    # Base Classes
+    'Interaction',
+    'FunctionalInteraction',
+    'TranslationalInteraction',
+    # Concrete Classes
+    'ComplExInteraction',
+    'ConvEInteraction',
+    'ConvKBInteraction',
+    'DistMultInteraction',
+    'ERMLPInteraction',
+    'ERMLPEInteraction',
+    'HolEInteraction',
+    'KG2EInteraction',
+    'NTNInteraction',
+    'ProjEInteraction',
+    'RESCALInteraction',
+    'RotatEInteraction',
+    'SimplEInteraction',
+    'StructuredEmbeddingInteraction',
+    'TransDInteraction',
+    'TransEInteraction',
+    'TransHInteraction',
+    'TransRInteraction',
+    'TuckerInteraction',
+    'UnstructuredModelInteraction',
+]
+
+logger = logging.getLogger(__name__)
+
+
+def _get_batches(z, slice_size):
+    for batch in zip(*(hh.split(slice_size, dim=1) for hh in ensure_tuple(z)[0])):
+        if len(batch) == 1:
+            batch = batch[0]
+        yield batch
+
+
+# TODO rename Interaction -> FunctionalInteraction and then BaseInteraction -> Interaction (later)
```

Comment:
shall we fix this one now? Otherwise It will introduce changes in the superclass of all subclasses later on

Reply:
oh i already did it. just forgot to deletee the todo

Reply:
Fixed in 7f18311d